.github/workflows/unit-tests.yaml

```
@@ -0,0 +1,54 @@
+concurrency:
+  group: pr-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+    setup-and-run:
+      runs-on: ubuntu-latest
+      timeout-minutes: 60
+  
+      steps:
+        - name: Checkout repository
+          uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)